core/pkg/store/query.go

```diff
@@ -28,15 +28,24 @@ const flagSetIdKeySourceCompoundIndex = flagSetIdIndex + "+" + keyIndex + "+" +
 var nilFlagSetId = uuid.New().String()
 
 // A selector represents a set of constraints used to query the store.
-type Selector struct {
+type Selector interface {
```

Comment:
I made this an interface because I was passing it as a pointer everywhere anyway.

Reply:
I reverted this.